File: app/src/main/java/com/pierdr/pierluigidallarosa/myactivity/ConsoleManager.java
package com.pierdr.pierluigidallarosa.myactivity;

import android.widget.TextView;


public class ConsoleManager {
    public TextView c1;

    private String[] messages;
    ConsoleManager(TextView c1)
    {
        this.c1 = c1;

        messages = new String[3];
        for(int i = 0;i<3;i++)
        {
            messages[i]="";
        }
    }
    public void addNewMessage(String newMessage)
    {
        messages[2] = messages[1];
        messages[1] = messages[0];
        messages[0]=newMessage;
        c1.setText(String.format("-> %s\n->%s\n->%s", messages[0], messages[1], messages[2]));
    }


}

File: app/src/main/java/com/pierdr/pierluigidallarosa/myactivity/ConsoleManager.kt
package com.pierdr.pierluigidallarosa.myactivity

import android.widget.TextView


internal class ConsoleManager(private val c1: TextView) {

    private val messages = MutableList(3) {""}

    fun addNewMessage(newMessage: String) {
        messages[2] = messages[1]
        messages[1] = messages[0]
        messages[0] = newMessage
        c1.text = String.format("-> %s\n->%s\n->%s", messages[0], messages[1], messages[2])
    }


}
